File: IPWorks MQ Samples/Amazon SQS Queue Manager/javascript/sqs.js
/*
 * IPWorks MQ 2024 JavaScript Edition - Sample Project
 *
 * This sample project demonstrates the usage of IPWorks MQ in a 
 * simple, straightforward way. It is not intended to be a complete 
 * application. Error handling and other checks are simplified for clarity.
 *
 * www.nsoftware.com/ipworksmq
 *
 * This code is subject to the terms and conditions specified in the 
 * corresponding product license agreement which outlines the authorized 
 * usage and restrictions.
 */
 
const readline = require("readline");
const ipworksmq = require("@nsoftware/ipworksmq");

if(!ipworksmq) {
  console.error("Cannot find ipworksmq.");
  process.exit(1);
}
let rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});



const promt = (query) => new Promise((resolve) => rl.question(query, resolve));

var queue = new ipworksmq.amazonsqs();
var myQueueId = "";

queue.on('Queue', function(e)
{
  console.log("Queue ID: " + e.queueId);
  console.log("URL" + e.URL);
});

queue.on("Message", function(e) 
{
  console.log("Message ID: " + e.messageId);
  console.log("Message data: " + e.messageData);
  console.log("Receipt handle: " + e.receiptHandle);
});

main();

function printMenu()
{
  console.log('\r\n?\t-\tHelp\n' +
    'cd\t-\tSelect Queue\n' +
    'del\t-\tDelete Queue\n' +
    'mk\t-\tCreate Queue\n' +
    'ls\t-\tList Queues\n' +
    'lsmsg\t-\tList Messages\n' +
    'delmsg\t-\tDelete Message\n' +
    'mkmsg\t-\tCreate Message\n' +
    'q\t-\tQuit\n');
}

async function main() {
  queue.setAccessKey(await promt("AWS Access Key: "));
  queue.setSecretKey(await promt("AWS Secret Key: "));
  
  printMenu();
  console.log("\nAvailable queues:");
  queue.listQueues();

  setTimeout(function () 
  {
    process.stdout.write("\nEnter command: ");
  }, 1000);

  rl.on("line", async function (line)
  {
    switch (line.toLowerCase())
    {
      case "cd":
        myQueueId = await promt("Enter queue ID: ");
        break;
          
      case "del":
        await queue.deleteQueue(await promt("Delete queue with ID: ")).catch(function (e)
          {
            console.log(e.message);
          });
        break;
    
      case "mk":
        console.log("Queue created with Id: " + await queue.createQueue(await promt("Enter queue name: ")).catch(function (e)
          {
            console.log(e.message);
          }));
        break;
      
      case "ls":
        await queue.listQueues().catch(function (e)
          {
            console.log(e.message);
          });
        break;
      
      case "lsmsg":
        await queue.listMessages(myQueueId).catch(function (e)
          {
            console.log(e.message);
          });
        break;
      
      case "delmsg":
        await queue.deleteMessage(myQueueId, await promt("Enter receipt handle: ").catch(function (e)
          {
            console.log(e.message);
          }));
        break;
      
      case "mkmsg":
        console.log("Message created with ID: " + await queue.createMessage(myQueueId, await promt("Enter message data: ")).catch(function (e)
          {
            console.log(e.message);
          }));
        break;

      case "?":
        printMenu();
        break;

      case "q":
        process.exit(0);

      default:
        console.log("Command not recognized!");
        break;
    }
    process.stdout.write("\nEnter command: ");
  });
}

function prompt(promptName, label, punctuation, defaultVal)
{
  lastPrompt = promptName;
  lastDefault = defaultVal;
  process.stdout.write(`${label} [${defaultVal}] ${punctuation} `);
}

File: IPWorks MQ Samples/Amazon SQS Queue Manager/python/sqs.py
# 
# IPWorks MQ 2024 Python Edition - Sample Project
# 
# This sample project demonstrates the usage of IPWorks MQ in a 
# simple, straightforward way. It is not intended to be a complete 
# application. Error handling and other checks are simplified for clarity.
# 
# www.nsoftware.com/ipworksmq
# 
# This code is subject to the terms and conditions specified in the 
# corresponding product license agreement which outlines the authorized 
# usage and restrictions.
# 

import sys
import string
from ipworksmq import *

input = sys.hexversion<0x03000000 and raw_input or input

def ensureArg(args, prompt, index):
  if len(args) <= index:
    while len(args) <= index:
      args.append(None)
    args[index] = input(prompt)
  elif args[index] is None:
    args[index] = input(prompt)


queue = AmazonSQS()
command = ""
myQueueId = ""

def handle_queue_event(event: AmazonSQSQueueEventParams):
  print("Queue ID: " + event.queue_id)
  print("URL: " + event.url)

def handle_message_event(event: AmazonSQSMessageEventParams):
  print("Message ID: " + event.message_id)
  print("Message data: " + event.message_data)
  print("Meceipt handle: " + event.receipt_handle)

def printMenu():
  print("\r\n?\t-\tHelp\n" +
    "cd\t-\tSelect Queue\n" +
    "del\t-\tDelete Queue\n" +
    "mk\t-\tCreate Queue\n" +
    "ls\t-\tList Queues\n" +
    "lsmsg\t-\tList Messages\n" +
    "delmsg\t-\tDelete Message\n" +
    "mkmsg\t-\tCreate Message\n" +
    "q\t-\tQuit\n")

queue.on_queue = handle_queue_event
queue.on_message = handle_message_event

queue.set_access_key(input("AWS Access Key: "))
queue.set_secret_key(input("AWS Secret Key: "))

printMenu()
print("\nAvailable queues:")
queue.list_queues()

while True:
  command = input("\nEnter command: ")
  try:
    if command == "cd":
      myQueueId = input("Enter queue ID: ")
    elif command == "del":
      queue.delete_queue(input("Delete queue with ID: "))
    elif command == "mk":
      print("Queue created with ID: " + queue.create_queue(input("Enter queue name: ")))
    elif command == "ls":
      queue.list_queues()
    elif command == "lsmsg":
      queue.list_messages(myQueueId)
    elif command == "delmsg":
      queue.delete_message(myQueueId, input("Enter receipt handle: "))
    elif command == "mkmsg":
      print("Message created with ID: " + queue.create_message(myQueueId, input("Enter message data: ")))
    elif command == "q":
      sys.exit(0)
    elif command == "?":
      printMenu()
    else:
      print("Command not recognized!")
      printMenu()
  except IPWorksMQError as e:
      print("Error: " + e.message)
  except Exception:
      print(e.message)
      sys.exit(1)

